### PR TITLE
Add initial swagger spec generation for API docs

### PR DESCRIPTION
**Pull request by ldevulder**
Ref #321 

This adds swagger comments into a separate doc package, so they don't interfere with existing comments.  Models are scanned automatically, but routes are not and need to be added manually.

https://goswagger.io/use/spec.html

This is based on the closed PR #272.

### CODE REVIEW

File: internal/api/v1/docs/namespace.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+// swagger:route POST /namespaces/{namespace} namespace NamespaceCreate
+// Create a new named `namespace`.
+// responses:
+//   200: NamespaceCreateResponse
+
+// swagger:parameters NamespaceCreate
+type NamespaceCreateParam struct {
+	Namespace models.NamespaceCreateRequest
```
Comment (ldevulder) [inline internal/api/v1/docs/namespace.go:27]: I'm not sure about this one: should it be `string` instead of `models.NamespaceCreateRequest`?

Comment (andreas-kupries) [reply internal/api/v1/docs/namespace.go:27]: It should be `string`, and there should be a
```
// in: body
Body models.NamespaceCreateRequest
```
for the POST body.

File: internal/api/v1/docs/application.go [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+
+// swagger:parameters AppLogs
+type AppLogsParam struct {
+	Namespace string
+	App       string
+}
+
+// swagger:response AppLogsResponse
+type AppLogsResponse struct {
+	// in: body
```
Comment (ldevulder) [inline internal/api/v1/docs/application.go:81]: I haven't found what to add for the `LogsResponse`... Any hints?

File: internal/api/v1/docs/application.go [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+
+// swagger:parameters StagingLogs
+type StagingLogsParam struct {
+	Namespace string
+	StageId   string
+}
+
+// swagger:response StagingLogsResponse
+type StagingLogsResponse struct {
+	// in: body
```
Comment (ldevulder) [inline internal/api/v1/docs/application.go:97]: I haven't found what to add for the `LogsResponse`... Any hints?

File: internal/api/v1/docs/application.go (hunk 2 above)
Comment (andreas-kupries) [reply internal/api/v1/docs/application.go:81]: Heh. The log endpoints upgrade the connection to a websocket and then transfer log entries over it. They do not have a regular response per se I believe.

A very short google does not net me info on how to describe this for swagger. ... If you have more luck let us use that. And if not I would leave the response off, and make note of the websocket upgrade in the endpoint description.

File: internal/api/v1/docs/application.go (hunk 3 above)
Comment (andreas-kupries) [reply internal/api/v1/docs/application.go:97]: See above.

File: internal/api/v1/docs/application.go (hunk 2 above)
Comment (ldevulder) [reply internal/api/v1/docs/application.go:81]: Yes, it seems that swagger is not able to describe websocket connection.

So I can change the endpoint description to something like this: `Return logs of the named 'application' in the 'namespace' streamed over a websocket.`
WDYT?

Comment (ldevulder) [reply internal/api/v1/docs/application.go:81]: I've pushed a comment change, let me know if this is not OK for you.

Comment (andreas-kupries) [reply internal/api/v1/docs/application.go:87]: I am :+1: on that change.

**Review by jimmykarily — APPROVED** [review]

File: Makefile [hunk 4]
```diff
@@ -147,6 +147,18 @@ embed_files: getstatik wrap_registry_chart
 help:
 	( echo _ _ ___ _____ ________ Overview ; epinio help ; for cmd in apps completion create-org delete help info install orgs push target uninstall ; do echo ; echo _ _ ___ _____ ________ Command $$cmd ; epinio $$cmd --help ; done ; echo ) | tee HELP
 
+
+########################################################################
+# Docs
+
+swagger:
+	swagger generate spec > docs/references/api/swagger.json
```
[the remark this answers is not in the document]
Comment (andreas-kupries) [reply Makefile:155]: I use a binary from https://github.com/go-swagger/go-swagger/releases

[the remark this answers is not in the document]
Comment (andreas-kupries) [reply Makefile:154]: Done. Commit c776ee0a6d909ed4bcab960db8b661c264f8d84a

**Review by jimmykarily — APPROVED** [review]

**Pull request merged**